pkg/chains/formats/slsa/internal/results/results.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2024 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package results
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/tektoncd/chains/pkg/artifacts"
+	"github.com/tektoncd/chains/pkg/chains/objects"
+
+	slsa "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v1"
+)
+
+// GetResultsWithoutBuildArtifacts returns all the results without those that are build artifacts.
+func GetResultsWithoutBuildArtifacts(results []objects.Result, resultTypePrefix string) ([]slsa.ResourceDescriptor, error) {
+	byProd := []slsa.ResourceDescriptor{}
+	for _, r := range results {
+		if isBuildArt, err := artifacts.IsBuildArtifact(r); err == nil && isBuildArt {
```

Comment:
What if err is not nil? If that doesn't matter here, let's consider rewriting it to something like:

```go
if isBuildArt, _ := artifacts.IsBuildArtifact(r); isBuildArt {
```